src/mds/MDSRank.cc

```diff
@@ -1760,13 +1760,16 @@ bool MDSRankDispatcher::handle_asok_command(
   } else if (command == "session evict") {
     std::string client_id;
     const bool got_arg = cmd_getval(g_ceph_context, cmdmap, "client_id", client_id);
-    assert(got_arg == true);
+    if(!got_arg) {
+      ss << "Invalid client_id specified";
+      return true;
+    }
 
     mds_lock.Lock();
-    std::stringstream ss;
-    bool killed = kill_session(strtol(client_id.c_str(), 0, 10), true, ss);
+    std::stringstream dss;
```

Comment:
I think the right thing here would be to just remove the `ss` declaration that shadows the outer one.  That way the messages that come out of ``kill_session`` can bubble up to the user.